src/test/localization/WIP_all.test.js

```diff
@@ -0,0 +1,137 @@
+  let pathsArr = [];
+  
+  // Arrays in localization are interpolated into translated strings, so don't do anything
+  if (Array.isArray(obj)) {
+    return pathsArr;
+
+  // Return an array of paths to each prop using the base path and the key in each loop iteration
+  } else if (typeof obj === 'object') {
+    const keys = Object.keys(obj).sort();
+    
+    // Handle having an empty object
+    if (keys.length === 0) {
+      return pathsArr;
+
+    // Otherwise, loop through the keys and recursively call compare()
+    } else {
+      for (let key of keys) {
+        // Append this key to the base path array
+        let newBase = [ 
+          ...base,
+          ...[ key ],
+        ];
+        pathsArr.push(newBase);
+
+        // Pass this object back to our function to get paths to any children
+        let childPaths = getKeyPathsArray(obj[ key ], newBase);
+        
+        // Add any child paths to our array before returning
+        pathsArr = pathsArr.concat(childPaths);
+      }
+      return pathsArr;
+    }
+
+  // Otherwise, we've got some sort of primative such as a str, num, etc, so don't do anything
+  } else {
+    return pathsArr;
+  }
+};
+
+
+// Get the list of non-EN localizations to test
+let localizationKeys = Object.keys(localizations);
+const enIndex = localizationKeys.indexOf('en');
+
+// Only run our comparisons if EN localization exists
+if (enIndex !== -1) {
+
+  const keyPaths = {};
+  for (let key of localizationKeys) {
+    // Get the key paths from the localization files
+    keyPaths[ key ] = getKeyPathsArray(localizations[ key ]);
+  }
+
+  // Remove EN from our keys to loop over
+  localizationKeys.splice(enIndex, 1);
+  
+  // Loop through all non-en localizations
+  for (let i = 0; i < localizationKeys.length; i++) {
+    let locKey = localizationKeys[ i ];
+   
+    describe(`localization ${locKey}`, () => {
+    
+      // Test for keys in 'EN' that aren't localization being tested
+      for (let j = 0; j < keyPaths[ 'en' ].length; j++) {
+        let keyPath = keyPaths[ 'en' ][ j ];
+        let keyPathAsStr = keyPath.join('.');
+        let keyExistsInLoc = _.has(localizations[ locKey ], keyPath);
+
+        it(`'en' path ${keyPathAsStr} exists in '${locKey}'`, () => {
+          expect(keyExistsInLoc).toBe(true);
+        });
+
+        if (keyExistsInLoc) {
+          // Get the value of our key so we can test its type
+          let locVal = _.get(localizations[ locKey ], keyPath);
+          
+          // We'll only compare the values of strings and arrays.
+          // Use a simple deep compare since arrays eventually 
+          // become strings and we want to know if the text elements
+          // of the array have been translated
+          if (typeof locVal !== 'object' || Array.isArray(locVal)) {
+            it(`${keyPathAsStr} translation should not be the same as '${locKey}'`, () => {
+              expect(_.isEqual(locVal, _.get(localizations[ 'en' ], keyPath))).not.toBe(true);
```

Comment:
Sorry about this - you've talked about this aspect of the check for a while I've felt there was something off about that aspect of it, but I couldn't put my finger on what it was. Now I've remembered.

An easy example is Spanish - 'no' in Spanish is also 'no'. In a more complex example, German is apparently very strict about using a language's own name for its pronouns. So, 'Cliff Effects Tool' stays 'Cliff Effects Tool'. That part never gets changed.

Reply:
Hmm, I was wondering if that sort of thing would crop up.  I saw things like "Cliff Effects Werkzeug" and "Cliff Effects Herriamienta" and was hoping instances not changed were WIP. This is a tough one then.  We could simply not test values, but then we'd lose a big benefit of this comparison.  Any thoughts on that?  Some indicator like a suffix to a key that indicates not to expect a translation?  Or some other metadata to indicate that?

Reply:
That's what version names are. `key_v#`.

Reply:
In the google docs

Reply:
We're working on bringing the version numbers into the local docs.

Reply:
I believe I was tripped up by the app name being in a couple places in the loc files and sometimes it was translated and sometimes not.  So I was trying to do string comparison to catch places where an English phase was incorrectly being used as a placeholder.  Sounds like the manual process of versioning should be catching that and any instances of english should be there.  So I can simply remove my string comparison and we're good on that front?  Just check the structure of the doc?

Reply:
Yeah, I think that's what I was saying. For some reason the key name check hasn't been clicking in my head. Yup, just don't check for a string-to-string match and that sounds good!

Reply:
I've made a separate pull request for the localization report since this won't live on in the form of tests.  That includes removing the value comparison, and just leaves the key checks.  I think we can close this without merge at this point and pick things up in that one?  #757

Reply:
Sounds like a plan! Feel free to close when you feel comfortable!